crates/runtime/src/dataconnector/mongodb.rs

```diff
@@ -0,0 +1,204 @@
+    }
+}
+
+#[derive(Default, Copy, Clone)]
+pub struct MongoDBFactory {}
+
+impl MongoDBFactory {
+    #[must_use]
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    #[must_use]
+    pub fn new_arc() -> Arc<dyn DataConnectorFactory> {
+        Arc::new(Self {}) as Arc<dyn DataConnectorFactory>
+    }
+}
+
+const PARAMETERS: &[ParameterSpec] = &[
+    ParameterSpec::component("connection_string").secret(),
+    ParameterSpec::component("user").secret(),
+    ParameterSpec::component("pass").secret(),
+    ParameterSpec::component("host"),
+    ParameterSpec::component("port"),
+    ParameterSpec::component("db"),
+    ParameterSpec::component("sslmode"),
+    ParameterSpec::component("sslrootcert"),
+    ParameterSpec::component("auth_source"),
+    ParameterSpec::component("direct_connection"),
+    ParameterSpec::component("pool_min")
+        .description("The minimum number of connections to keep open in the pool, lazily created when requested.")
+        .default("0"),
+    ParameterSpec::component("pool_max")
+        .description("The maximum number of connections to allow in the pool.")
+        .default("10"),
+];
+
+impl DataConnectorFactory for MongoDBFactory {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn create(
+        &self,
+        mut params: ConnectorParams,
+    ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
+        Box::pin(async move {
+            let mut pool_min = params
+                .parameters
+                .get("pool_min")
+                .ok()
+                .and_then(|s| {
+                    let pool_min_str = s.expose_secret();
+                    let parsed_pool_min = pool_min_str.parse::<usize>();
+                    if parsed_pool_min.is_err() {
+                        tracing::warn!(
+                            "Invalid pool_min value: {pool_min_str}, using default of 10"
+                        );
+                    }
+                    parsed_pool_min.ok()
+                })
+                .unwrap_or(10);
+            let mut pool_max = params
+                .parameters
+                .get("pool_max")
+                .ok()
+                .and_then(|s| {
+                    let pool_max_str = s.expose_secret();
+                    let parsed_pool_max = pool_max_str.parse::<usize>();
+                    if parsed_pool_max.is_err() {
+                        tracing::warn!(
+                            "Invalid pool_max value: {pool_max_str}, using default of 100"
+                        );
+                    }
+                    parsed_pool_max.ok()
+                })
+                .unwrap_or(100);
+
+            if pool_min > pool_max {
+                tracing::warn!(
+                    "pool_min value: {pool_min} is greater than pool_max value: {pool_max}, using default values of 10 and 100"
+                );
+                pool_min = 10;
+                pool_max = 100;
```

Comment:
Can we put these magic numbers in a single `const` location (i.e. `const DEFAULT_POOL_MIN: usize = 10`), since we refer to them a couple times?